mypyc/test-data/run-strings.test

```diff
@@ -149,6 +149,14 @@ def test_str_to_bool() -> None:
         assert is_true(x)
         assert not is_false(x)
 
+def test_str_join() -> None:
+    var = 'mypyc'
+    num = 10
+    assert ''.join(['a', 'b', '{}'.format(var), 'c']) == 'abmypycc'
+    assert ''.join(['a', 'b', '{:{}}'.format(var, ''), 'c']) == 'abmypycc'
+    assert ''.join(['a', 'b', '{:{}}'.format(var, '>10'), 'c']) == 'ab     mypycc'
+    assert ''.join(['a', 'b', '{:{}}'.format(var, '>{}'.format(num)), 'c']) == 'ab     mypycc'
```

Comment:
Ideas for a few more test cases: `x.join(...)` (variable `x`), `','.join(...)` (non-empty string), `''.join(['x', v])` (variable in argument).